### PR TITLE
Refactor balo generation code

### PR DESCRIPTION
**Pull request by NatashaWso2**
- deletes the existing balo before creating a new one
- ignores folders which are not packages when building a ballerina package

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/FileSystemProjectDirectory.java [hunk 1]
```diff
@@ -90,7 +90,10 @@ public Path getPath() {
     private boolean isSpecialDirectory(Path dirName) {
         // TODO Improve this logic in an extensible manner.
         return dirName.toString().equals(ProjectDirConstants.DOT_BALLERINA_DIR_NAME) ||
-                dirName.toString().equals(ProjectDirConstants.TARGET_DIR_NAME);
+                dirName.toString().equals(ProjectDirConstants.TARGET_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.RESOURCE_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.TEST_DIR_NAME) ||
+                dirName.toString().equals(ProjectDirConstants.DOT_GIT_DIR_NAME);
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/FileSystemProjectDirectory.java]: this has unnecessary repetition, also this list can be dynamic in the future. let's have a list.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java [hunk 2]
```diff
@@ -59,38 +58,52 @@ static void generateBalo(BLangPackage bLangPackage, String projectPath, Stream<P
             try {
                 Files.createDirectories(destPath);
             } catch (IOException e) {
-                throw new BLangCompilerException("Error occurred when creating directories in " +
-                                                         "./ballerina/repo/ to save the generated balo");
+                throw new BLangCompilerException("error creating directories in ./ballerina/repo/ to save the " +
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java]: This is a temp hack I did, can you please improve using createAndGetHomeReposPath()

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/FileSystemProjectDirectory.java (hunk 1 above)
Comment (NatashaWso2) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/FileSystemProjectDirectory.java]: Fixed with c1901bb

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java [hunk 3]
```diff
@@ -59,38 +58,52 @@ static void generateBalo(BLangPackage bLangPackage, String projectPath, Stream<P
...
+                                                         "generated balo");
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java]: shouldn't say balo to user.

Comment (NatashaWso2) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/ZipUtils.java]: Changed balo to artifact : aee8b8a68a74934a7076d0ebd12ba1592eeadd16

**Pull request merged**